fs/fs.go

```diff
@@ -81,13 +82,66 @@ var (
 	fusermountBin               = "fusermount"
 )
 
+type layerQueue struct {
+	mu       sync.Mutex
+	queue    chan func() string
+	cache    *hashset.Set
+	maxSize  int
+	currSize int
+}
+
+func (lq *layerQueue) Start() error {
+	if lq.queue != nil {
+		return fmt.Errorf("layer queue already instantiated")
+	}
+
+	lq.queue = make(chan func() string)
```

Comment:
This channel needs to be buffered or else all enqueue operations will block when we're at our concurrency limit.